modules/ROOT/pages/migration/index.adoc

```diff
@@ -1,729 +1,1008 @@
 
-[[migration-guide-server]]
-== Server
+Here is a list of all the breaking changes from version 3.0.0 to 4.0.0.
 
-[[migration-guide-server-schema-generation]]
-=== Schema Generation
+=== `IExecutableSchemaDefinition`
 
-In your server codebase, the process of creating an executable schema has changed. For a simple application, what used to be:
+If you were passing any arguments from https://the-guild.dev/graphql/tools/docs/api/interfaces/schema_src.iexecutableschemadefinition[`IExecutableSchemaDefinition`] into the library other than `typeDefs` and `resolvers`, these are no longer supported.
 
-[source, javascript, indent=0]
-----
-const { makeAugmentedSchema } = require("neo4j-graphql-js");
+=== `config.enableDebug`
 
-const typeDefs = require("./type-definitions");
+The programmatic toggle for debug logging has been moved from `config.enableDebug` to simply `debug`.
 
-const schema = makeAugmentedSchema({ typeDefs });
-----
+[cols="1,1"]
+|===
+|Before | Now
 
-Has become:
+a|
+[source, javascript, indent=0]
+----
+const { Neo4jGraphQL } = require("@neo4j/graphql");
+const neo4j = require("neo4j-driver");
+const { ApolloServer } = require("apollo-server");
 
+const typeDefs = `
+    type Movie {
+        title: String!
+    }
+`;
+
+const driver = neo4j.driver(
+    "bolt://localhost:7687",
+    neo4j.auth.basic("neo4j", "password")
+);
+
+const neoSchema = new Neo4jGraphQL({
+    typeDefs,
+    driver,
+    config: {
+      enableDebug: true,
+    }
+});
+----
+a|
 [source, javascript, indent=0]
 ----
 const { Neo4jGraphQL } = require("@neo4j/graphql");
+const neo4j = require("neo4j-driver");
+const { ApolloServer } = require("apollo-server");
 
-const typeDefs = require("./type-definitions");
+const typeDefs = `
+    type Movie {
+        title: String!
+    }
+`;
 
-const neo4jGraphQL = new Neo4jGraphQL({ typeDefs });
+const driver = neo4j.driver(
+    "bolt://localhost:7687",
+    neo4j.auth.basic("neo4j", "password")
+);
 
-const schema = await neo4jGraphQL.getSchema();
+const neoSchema = new Neo4jGraphQL({
+    typeDefs,
+    driver,
+    debug: true,
+});
 ----
+|===
 
-Additionally, the `Neo4jGraphQL` constructor allows you to pass in a driver instance instead of passing one in the context of each request.
-
-=== Schema Configuration
+=== `driverConfig`
 
-The `neo4j-graphql-js` library would allow a user to configure the GraphQL schema via the `makeAugmentedSchema()` function.
+Session configuration is now available only in the context under the `sessionConfig` key.
+Additionally, the `bookmarks` key has been removed as it is no longer needed with the bookmark manager of the newer driver.
 
-For example, to exclude all queries and all mutations:
+[cols="1,1"]
+|===
+|Before | Now
 
+a|
 [source, javascript, indent=0]
 ----
-const schema = makeAugmentedSchema({
-  typeDefs,
-  config: {
-    query: false, // exclude all queries
-    mutation: false // exclude all mutations
-  }
-});
+const neoSchema = new Neo4jGraphQL({
```

Comment:
I believe here we comparing the before part only the `Neo4jGraphQL` initialization, but then in the `now` the whole configuration including imports, typeDefs, the server, etc... I would change the before part to be:

```typescript
import { ApolloServer } from '@apollo/server';
import { startStandaloneServer } from '@apollo/server/standalone';
import { Neo4jGraphQL } from "@neo4j/graphql";
import neo4j from "neo4j-driver";

const typeDefs = `#graphql
    type User {
        name: String
    }
`;

const driver = neo4j.driver(
    "bolt://localhost:7687",
    neo4j.auth.basic("neo4j", "password")
);

const neoSchema = new Neo4jGraphQL({
    typeDefs,
    config: {
        driverConfig: {
            database: "different-db"
        },
    },
})

const server = new ApolloServer({
    schema: await neoSchema.getSchema(),
});

await startStandaloneServer(server, {
    context: async ({ req }) => ({ req }),
});
```